src/nodes/Image.tsx

```diff
@@ -226,51 +253,92 @@ export default class Image extends Node {
   };
 
   toMarkdown(state, node) {
-    state.write(
+    let markdown =
       "![" +
-        state.esc((node.attrs.alt || "").replace("\n", "") || "") +
-        "](" +
-        state.esc(node.attrs.src) +
-        ")"
-    );
+      state.esc((node.attrs.alt || "").replace("\n", "") || "") +
+      "](" +
+      state.esc(node.attrs.src);
+    if (node.attrs.layoutClass) {
+      markdown += ' "' + state.esc(node.attrs.layoutClass) + '"';
+    }
+    if (node.attrs.title) {
+      markdown += ' "' + state.esc(node.attrs.title) + '"';
+    }
+    markdown += ")";
+    state.write(markdown);
   }
 
   parseMarkdown() {
     return {
       node: "image",
-      getAttrs: token => ({
-        src: token.attrGet("src"),
-        alt: (token.children[0] && token.children[0].content) || null,
-      }),
+      getAttrs: token => {
+        return {
+          src: token.attrGet("src"),
+          alt: (token.children[0] && token.children[0].content) || null,
+          ...getLayoutAndTitle(token.attrGet("title")),
+        };
+      },
     };
   }
 
   commands({ type }) {
-    return attrs => (state, dispatch) => {
-      const { selection } = state;
-      const position = selection.$cursor
-        ? selection.$cursor.pos
-        : selection.$to.pos;
-      const node = type.create(attrs);
-      const transaction = state.tr.insert(position, node);
-      dispatch(transaction);
-      return true;
+    return {
+      deleteImage: () => (state, dispatch) => {
+        dispatch(state.tr.deleteSelection());
+        return true;
+      },
+      alignRight: () => (state, dispatch) => {
+        const attrs = {
+          ...state.selection.node.attrs,
+          title: null,
+          layoutClass: "half-right",
+        };
+        dispatch(state.tr.replaceSelectionWith(type.create(attrs)));
```

Comment:
We might consider using `tr.setNodeMarkup` to update the attribute of the existing image node rather than replacing it entirely for these, it'll be a bit more semantically correct, there are a couple of examples you can see of that in the codebase.

Reply:
Ok, I'll find an example and try that.